### DISCUSSION TITLE
Best practice: one mutation per table row?

### DISCUSSION THREAD

**Discussion by jesseduffield** [TanStack/query#7464] [Q&A] [answered]
I have a few tables in my app where each row has a button to perform an action (i.e. a mutation) and each row needs to show a loading state if the mutation is in flight. It seems like there are two ways to achieve this:
1) define a mutation in the parent component which takes a row identifier as an argument, along with an array of which rows are currently loading
2) define a mutation within each row component (which takes no arguments because it closes over the row identifier); no need for a centralised store of loading states

Option 2 is much easier to implement because I get per-row loading states for free, but I'm wondering if there's some good reason to prefer option 1 e.g. performance, separation of concerns.

Thanks

**Comment by TkDodo** [comment, marked as answer]
nah, option 2 is better.

**Reply by jesseduffield** [reply]
Thanks, that is exactly what I wanted to hear :)
